### PR TITLE
BUG: Using boolean keys to select a column (GH44322)

### LINKED ISSUE #44322: BUG: can't select single column from DataFrame by boolean key
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
print(pd.Series([1], index=pd.Index([False])).loc[False]) # Works i.e. returns 1
print(pd.DataFrame([[1]], index=pd.Index([False])).loc[False]) # Works i.e. returns [1]
print(pd.DataFrame([[1]], columns=pd.Index([False])).loc[:, False]) # Fails
```

### Issue Description

You can select a single *row* from a `DataFrame` by supplying a boolean key to `loc`, but not a single column. This seems inconsistent.

### Expected Behavior

Return the relevant column

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : 945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5
python           : 3.8.12.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18363
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United Kingdom.1252

pandas           : 1.3.4
numpy            : 1.21.3
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.1.1
setuptools       : 58.0.4
Cython           : 0.29.24
pytest           : 6.2.4
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.2
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
fsspec           : 2021.08.1
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : 3.6.1
tabulate         : None
xarray           : 0.19.0
xlrd             : 2.0.1
xlwt             : None
numba            : 0.53.1
</details>

### PR DESCRIPTION
**Pull request by DriesSchaumont**
- [x] closes #44322
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -178,6 +178,13 @@ def test_column_types_consistent(self):
         )
         tm.assert_frame_equal(df, expected)
 
+    def test_loc_getitem_column_boolean_arg(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:181]: can you add all of the examples from the OP (working & not-working), parameterizing if appropriate

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by phofl — COMMENTED** [review]
lgtm apart from @jreback's comment

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (DriesSchaumont) [reply pandas/tests/indexing/test_loc.py]: Done. Added `isinstance` check because `tm.assert_equal` raises `NotImplementedError` for scalars.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @DriesSchaumont

**Pull request merged**